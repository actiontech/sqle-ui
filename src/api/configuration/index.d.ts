import {
  IGetDriversResV1,
  IGetLDAPConfigurationResV1,
  ILDAPConfigurationReqV1,
  IBaseRes,
  IGetSMTPConfigurationResV1,
  IUpdateSMTPConfigurationReqV1,
  IGetSystemVariablesResV1,
  IUpdateSystemVariablesReqV1
} from '../common.d';

export interface IGetDriversV1Return extends IGetDriversResV1 {}

export interface IGetLDAPConfigurationV1Return
  extends IGetLDAPConfigurationResV1 {}

export interface IUpdateLDAPConfigurationV1Params
  extends ILDAPConfigurationReqV1 {}

export interface IUpdateLDAPConfigurationV1Return extends IBaseRes {}

export interface IGetSMTPConfigurationV1Return
  extends IGetSMTPConfigurationResV1 {}

export interface IUpdateSMTPConfigurationV1Params
  extends IUpdateSMTPConfigurationReqV1 {}

export interface IUpdateSMTPConfigurationV1Return extends IBaseRes {}

export interface IGetSystemVariablesV1Return extends IGetSystemVariablesResV1 {}

export interface IUpdateSystemVariablesV1Params
  extends IUpdateSystemVariablesReqV1 {}

export interface IUpdateSystemVariablesV1Return extends IBaseRes {}
