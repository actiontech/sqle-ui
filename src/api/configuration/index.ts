/* tslint:disable no-identical-functions */
/* tslint:disable no-useless-cast */
/* tslint:disable no-unnecessary-type-assertion */
/* tslint:disable no-big-function  */
/* tslint:disable no-duplicate-string  */
import ServiceBase from '../Service.base';
import { AxiosRequestConfig } from 'axios';

import {
  IGetDriversV1Return,
  IGetLDAPConfigurationV1Return,
  IUpdateLDAPConfigurationV1Params,
  IUpdateLDAPConfigurationV1Return,
  IGetSMTPConfigurationV1Return,
  IUpdateSMTPConfigurationV1Params,
  IUpdateSMTPConfigurationV1Return,
  IGetSystemVariablesV1Return,
  IUpdateSystemVariablesV1Params,
  IUpdateSystemVariablesV1Return
} from './index.d';

class ConfigurationService extends ServiceBase {
  public getDriversV1(options?: AxiosRequestConfig) {
    return this.get<IGetDriversV1Return>(
      '/v1/configurations/drivers',
      undefined,
      options
    );
  }

  public getLDAPConfigurationV1(options?: AxiosRequestConfig) {
    return this.get<IGetLDAPConfigurationV1Return>(
      '/v1/configurations/ldap',
      undefined,
      options
    );
  }

  public updateLDAPConfigurationV1(
    params: IUpdateLDAPConfigurationV1Params,
    options?: AxiosRequestConfig
  ) {
    const paramsData = this.cloneDeep(params);
    return this.patch<IUpdateLDAPConfigurationV1Return>(
      '/v1/configurations/ldap',
      paramsData,
      options
    );
  }

  public getSMTPConfigurationV1(options?: AxiosRequestConfig) {
    return this.get<IGetSMTPConfigurationV1Return>(
      '/v1/configurations/smtp',
      undefined,
      options
    );
  }

  public updateSMTPConfigurationV1(
    params: IUpdateSMTPConfigurationV1Params,
    options?: AxiosRequestConfig
  ) {
    const paramsData = this.cloneDeep(params);
    return this.patch<IUpdateSMTPConfigurationV1Return>(
      '/v1/configurations/smtp',
      paramsData,
      options
    );
  }

  public getSystemVariablesV1(options?: AxiosRequestConfig) {
    return this.get<IGetSystemVariablesV1Return>(
      '/v1/configurations/system_variables',
      undefined,
      options
    );
  }

  public updateSystemVariablesV1(
    params: IUpdateSystemVariablesV1Params,
    options?: AxiosRequestConfig
  ) {
    const paramsData = this.cloneDeep(params);
    return this.patch<IUpdateSystemVariablesV1Return>(
      '/v1/configurations/system_variables',
      paramsData,
      options
    );
  }
}

export default new ConfigurationService();
