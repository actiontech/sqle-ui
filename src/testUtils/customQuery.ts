import * as domTestingLib from '@testing-library/dom';
import { fireEvent, screen } from '@testing-library/react';

export const getBySelector = (selector: string, baseElement?: Element) => {
  let temp;
  if (baseElement !== undefined) {
    temp = baseElement.querySelectorAll(selector);
  } else {
    temp = document.querySelectorAll(selector);
  }
  if (temp.length === 0) {
    throw domTestingLib.queryHelpers.getElementError(
      `Unable to find an element by: selector=${selector}`,
      document.body
    );
  }
  if (temp.length > 1) {
    throw domTestingLib.queryHelpers.getElementError(
      `Found multiple elements with the selector=${selector}`,
      document.body
    );
  }
  return temp[0];
};

export const getAllBySelector = (selector: string, baseElement?: Element) => {
  let temp;
  if (baseElement !== undefined) {
    temp = baseElement.querySelectorAll(selector);
  } else {
    temp = document.querySelectorAll(selector);
  }
  if (temp.length === 0) {
    throw domTestingLib.queryHelpers.getElementError(
      `Unable to find an element by: selector=${selector}`,
      document.body
    );
  }
  return temp;
};

export const getSelectContentByFormLabel = (label: string) => {
  return getBySelector(
    '.ant-select-selection-item-content',
    screen.getByText(label).parentNode?.parentNode as HTMLDivElement
  );
};

export const getSelectValueByFormLabel = (label: string) => {
  return getBySelector(
    '.ant-select-selection-item',
    screen.getByLabelText(label).parentNode?.parentNode as HTMLDivElement
  );
};

/**
 * 模拟选择 Select 组件中下拉框中的某项数据
 * @param 表单中 Select 组件外部 Form.Item 的label, 注意该 Form.Item 必须有 name
 * @param optionText 需要选中的下拉项
 * @param index 通过文本获取所有的节点会有两个(具体信息可以在网页上查看Select 组件中下拉的节点信息), 正常来说其中第二个是需要来需要用户点击的,
 * 所以这里的 index 默认值为 1. 并且这个节点拥有类名 ant-select-item-option-content, 所以在点击前需要使用 expect 来判断是否为正确的节点.
 */
export const selectOptionByIndex = (
  label: string,
  optionText: string,
  index = 1
) => {
  fireEvent.mouseDown(screen.getByLabelText(label));
  const options = screen.getAllByText(optionText);
  let realIndex = index;
  if (index < 0) {
    realIndex = options.length + index;
  }
  const option = options[realIndex];
  expect(option).toHaveClass('ant-select-item-option-content');
  fireEvent.click(option);
};

/**
 * 具体功能同 selectOptionByIndex, 不过该函数主要针对下拉框选项为一个节点的情况
 * @param label 表单中 Select 组件外部 Form.Item 的label, 注意该 Form.Item 必须有 name
 * @param optionCls 下拉项节点最外层的 class
 * @param index  这里不同于 selectOptionByIndex 的 index, 其代表的就是真实需要选择的下拉项的序号, 从 0 开始.
 */
export const selectCustomOptionByClassName = (
  label: string,
  optionCls: string,
  index: number
) => {
  fireEvent.mouseDown(screen.getByLabelText(label));
  const options = getAllBySelector(`.${optionCls}`);
  let realIndex = index;
  if (index < 0) {
    realIndex = options.length + index;
  }

  const option = options[realIndex];
  expect(option.parentNode).toHaveClass('ant-select-item-option-content');
  fireEvent.click(option);
};

export const getHrefByText = (name: string) => {
  return screen.getByText(name).getAttribute('href');
};

export const getAllHrefByText = (name: string) => {
  return screen.getAllByText(name).map((v) => v.getAttribute('href'));
};
