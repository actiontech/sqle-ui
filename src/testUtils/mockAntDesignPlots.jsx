import { cloneDeep, isObject } from 'lodash';
const MockPlots = (props) => {
  const cloneProps = cloneDeep(props);

  Object.keys(cloneProps).forEach((key) => {
    if (isObject(cloneProps[key])) {
      cloneProps[key] = JSON.stringify(cloneProps[key]);
    }
  });
  return <div {...cloneProps} />;
};

const Line = MockPlots;
const Pie = MockPlots;
const RadialBar = MockPlots;
const Gauge = MockPlots;
const Column = MockPlots;
const mockRegisterShape = jest.fn();
const G2 = {
  registerShape: jest.fn(),
};

export { Line, Pie, RadialBar, Gauge, Column, G2, mockRegisterShape };

// eslint-disable-next-line import/no-anonymous-default-export
export default {
  Line,
  Pie,
  RadialBar,
  Gauge,
  Column,
  G2,
};
