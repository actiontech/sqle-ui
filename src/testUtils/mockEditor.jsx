const mockEditor = (props) => {
  // https://github.com/react-monaco-editor/react-monaco-editor/issues/176
  const { editorDidMount, ...otherProps } = props;
  return <input {...otherProps} />;
};

export default mockEditor;
