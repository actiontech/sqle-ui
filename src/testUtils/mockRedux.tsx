import { Dictionary } from '../types/common.type';
import * as redux from 'react-redux';
import locale from '../store/locale';
import user from '../store/user';
import userManage from '../store/userManage';
import whitelist from '../store/whitelist';
import nav from '../store/nav';
import { combineReducers, createStore } from 'redux';
import React from 'react';
import { Provider } from 'react-redux';

const reducers = combineReducers({
  locale,
  user,
  userManage,
  whitelist,
  nav,
});

export const storeFactory = (initStore: Dictionary = {}) => {
  return createStore(reducers, initStore);
};

export const mockUseSelector = (store?: any) => {
  const spy = jest.spyOn(redux, 'useSelector');
  spy.mockImplementation((getFn: Function) => {
    return getFn(store);
  });
  return spy;
};

export const mockUseDispatch = () => {
  const scopeDispatch = jest.fn();
  const spy = jest.spyOn(redux, 'useDispatch');
  spy.mockImplementation(() => scopeDispatch);
  return { spy, scopeDispatch };
};

export const mockUseLocation = () => {}


export const CustomProvider: React.FC<{ initStore: Dictionary }> = (props) => {
  return (
    <Provider store={storeFactory(props.initStore)}>{props.children}</Provider>
  );
};
