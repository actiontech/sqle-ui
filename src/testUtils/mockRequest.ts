import { AxiosResponse } from 'axios';
import instance from '../api/instance';
import role from '../api/role';
import ruleTemplate from '../api/rule_template';
import user from '../api/user';
import workflow from '../api/workflow';

export const successData = (data: any, otherData?: any) => {
  return {
    code: 0,
    message: '',
    data,
    ...otherData,
  };
};

export const failedData = (data?: any) => {
  return {
    code: 100,
    message: 'error',
    data,
  };
};

export const resolveImmediately = (
  data: any,
  { status = 200, headers = {}, config = {}, statusText = '' } = {}
) => {
  return Promise.resolve({
    status,
    headers,
    config,
    statusText,
    data: successData(data),
  });
};

export const rejectImmediately = (
  data: any,
  { status = 200, headers = {}, config = {}, statusText = '' } = {}
) => {
  return Promise.reject({
    status,
    headers,
    config,
    statusText,
    data: failedData(data),
  });
};

export const resolveThreeSecond = (
  data: any,
  {
    status = 200,
    headers = {},
    config = {},
    statusText = '',
    otherData = {},
  } = {}
) => {
  return new Promise<AxiosResponse<any>>((res) => {
    setTimeout(() => {
      res({
        status,
        headers,
        config,
        statusText,
        data: successData(data, otherData),
      });
    }, 3000);
  });
};

export const resolveErrorThreeSecond = (
  data: any,
  { status = 200, headers = {}, config = {}, statusText = '' } = {}
) => {
  return new Promise<AxiosResponse<any>>((res) => {
    setTimeout(() => {
      res({
        status,
        headers,
        config,
        statusText,
        data: failedData(data),
      });
    }, 3000);
  });
};

export const rejectThreeSecond = (
  data: any,
  { status = 200, headers = {}, config = {}, statusText = '' } = {}
) => {
  return new Promise<AxiosResponse<any>>((_, rej) => {
    setTimeout(() => {
      rej({
        status,
        headers,
        config,
        statusText,
        data: failedData(data),
      });
    }, 3000);
  });
};

export const throwErrorThreeSecond = (
  error = 'error',
  { status = 500, headers = {}, config = {}, statusText = '' } = {}
) => {
  return new Promise((res) => {
    throw new Error(error);
  });
};

export const mockUseInstance = () => {
  const spy = jest.spyOn(instance, 'getInstanceTipListV1');
  spy.mockImplementation(() =>
    resolveThreeSecond([{ instance_name: 'instance1' }])
  );
  return spy;
};

export const mockUseInstanceSchema = () => {
  const spy = jest.spyOn(instance, 'getInstanceSchemasV1');
  spy.mockImplementation(() =>
    resolveThreeSecond({ schema_name_list: ['schema1'] })
  );
  return spy;
};

export const mockUseRole = () => {
  const spy = jest.spyOn(role, 'getRoleTipListV1');
  spy.mockImplementation(() =>
    resolveThreeSecond([{ role_name: 'role_name1' }])
  );
  return spy;
};

export const mockUseRuleTemplate = () => {
  const spy = jest.spyOn(ruleTemplate, 'getRuleTemplateTipsV1');
  spy.mockImplementation(() =>
    resolveThreeSecond([{ rule_template_name: 'rule_template_name1' }])
  );
  return spy;
};

export const mockUseUsername = () => {
  const spy = jest.spyOn(user, 'getUserTipListV1');
  spy.mockImplementation(() =>
    resolveThreeSecond([{ user_name: 'user_name1' }])
  );
  return spy;
};

export const mockUseWorkflowTemplate = () => {
  const spy = jest.spyOn(workflow, 'getWorkflowTemplateTipsV1');
  spy.mockImplementation(() =>
    resolveThreeSecond([{ workflow_template_name: 'workflow-template-name-1' }])
  );
  return spy;
};
