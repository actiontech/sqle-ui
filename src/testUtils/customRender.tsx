import { render } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import {
  BrowserRouter,
  Router,
  RouterProps,
  MemoryRouter,
  MemoryRouterProps,
} from 'react-router-dom';
import { Provider } from 'react-redux';
import { Dictionary } from '../types/common.type';
import { storeFactory } from './mockRedux';
import lightTheme from '../theme/light';
import { ThemeProvider } from '@material-ui/styles';
import { mount, shallow } from 'enzyme';

type RenderParams = Parameters<typeof render>;
type MountParams = Parameters<typeof mount>;
type ShallowParams = Parameters<typeof shallow>;

export const renderWithRouter = (...[ui, option]: [...RenderParams]) => {
  return render(<BrowserRouter>{ui}</BrowserRouter>, option);
};

export const renderWithRedux = (
  ...[ui, option, initStore]: [...RenderParams, Dictionary?]
) => {
  return render(
    <Provider store={storeFactory(initStore)}>{ui}</Provider>,
    option
  );
};

export const renderHooksWithRedux = <TProps, TResult>(
  hooks: (props: TProps) => TResult,
  storeState: Dictionary = {}
) => {
  return renderHook(hooks, {
    wrapper: Provider,
    initialProps: {
      store: storeFactory(storeState),
    },
  } as any);
};

export const renderWithServerRouter = (
  ...[ui, option, props]: [...RenderParams, RouterProps]
) => {
  return render(<Router {...props}>{ui}</Router>, option);
};

export const renderWithMemoryRouter = (
  ...[ui, option, props]: [...RenderParams, MemoryRouterProps?]
) => {
  return render(<MemoryRouter {...props}>{ui}</MemoryRouter>, option);
};

export const renderWithTheme = (...[ui, option]: [...RenderParams]) => {
  return render(<ThemeProvider theme={lightTheme}>{ui}</ThemeProvider>, option);
};

export const renderWithThemeAndRouter = (
  ...[ui, option]: [...RenderParams]
) => {
  return render(
    <MemoryRouter>
      <ThemeProvider theme={lightTheme}>{ui}</ThemeProvider>
    </MemoryRouter>,
    option
  );
};

export const mountWithTheme = (...[ui, option]: [...MountParams]) => {
  return mount(ui, {
    ...option,
    wrappingComponent: ThemeProvider,
    wrappingComponentProps: {
      theme: lightTheme,
    },
  });
};

export const shallowWithTheme = (...[ui, option]: [...ShallowParams]) => {
  return shallow(ui, {
    ...option,
    wrappingComponent: ThemeProvider,
    wrappingComponentProps: {
      theme: lightTheme,
    },
  });
};
