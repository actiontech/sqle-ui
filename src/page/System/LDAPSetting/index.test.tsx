import { fireEvent, render, waitFor, screen } from '@testing-library/react';
import LdapConfig from '.';
import configuration from '../../../api/configuration';
import { resolveThreeSecond } from '../../../testUtils/mockRequest';

describe('System/LdapConfig', () => {
  beforeEach(() => {
    jest.useFakeTimers();
  });

  afterEach(() => {
    jest.useRealTimers();
    jest.clearAllMocks();
    jest.clearAllTimers();
  });

  const mockGetLdapConfig = () => {
    const spy = jest.spyOn(configuration, 'getLDAPConfigurationV1');
    spy.mockImplementation(() =>
      resolveThreeSecond({
        enable_ldap: false,
        ldap_server_host: '1',
        ldap_server_port: '1',
        ldap_connect_dn: '1',
        ldap_search_base_dn: '1',
        ldap_user_name_rdn_key: '1',
        ldap_user_email_rdn_key: '1',
      })
    );
    return spy;
  };

  const mockUpdateldapConfig = () => {
    const spy = jest.spyOn(configuration, 'updateLDAPConfigurationV1');
    spy.mockImplementation(() => resolveThreeSecond({}));
    return spy;
  };

  test('should get ldap setting from origin', async () => {
    mockGetLdapConfig();
    const { container } = render(<LdapConfig />);
    expect(container).toMatchSnapshot();
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    expect(container).toMatchSnapshot();
  });

  test('should update ldap setting after user update from', async () => {
    const getSpy = mockGetLdapConfig();
    const updateSpy = mockUpdateldapConfig();
    const { container } = render(<LdapConfig />);
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    fireEvent.click(screen.getByText('common.modify'));
    expect(container).toMatchSnapshot();
    fireEvent.click(screen.getByLabelText('system.ldap.enableLdap'));
    fireEvent.input(screen.getByLabelText('system.ldap.ldapServerPort'), {
      target: { value: '3306' },
    });

    fireEvent.input(screen.getByLabelText('system.ldap.ldapConnectPwd'), {
      target: { value: '123456' },
    });

    fireEvent.click(screen.getByText('common.submit'));
    await waitFor(() => {
      jest.advanceTimersByTime(0);
    });

    expect(updateSpy).toBeCalledTimes(1);
    expect(updateSpy).toBeCalledWith({
      enable_ldap: true,
      ldap_connect_dn: '1',
      ldap_search_base_dn: '1',
      ldap_server_host: '1',
      ldap_server_port: '3306',
      ldap_connect_pwd: '123456',
      ldap_user_email_rdn_key: '1',
      ldap_user_name_rdn_key: '1',
    });
    expect(screen.getByText('common.submit').parentNode).toHaveClass(
      'ant-btn-loading'
    );
    expect(screen.getByText('common.cancel').parentNode).toHaveAttribute(
      'disabled'
    );
    expect(getSpy).toBeCalledTimes(1);
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    expect(getSpy).toBeCalledTimes(2);
    expect(container).toMatchSnapshot();
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    expect(screen.getByText('common.submit').parentNode).not.toHaveClass(
      'ant-btn-loading'
    );
    expect(screen.getByText('common.cancel').parentNode).not.toHaveAttribute(
      'disabled'
    );
  });
});
