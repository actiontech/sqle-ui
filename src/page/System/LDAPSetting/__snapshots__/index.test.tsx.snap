// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`System/LdapConfig should get ldap setting from origin 1`] = `
<div>
  <div
    class="ant-card ant-card-loading ant-card-bordered"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          system.title.ldap
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <div
        class="ant-skeleton ant-skeleton-active"
      >
        <div
          class="ant-skeleton-content"
        >
          <ul
            class="ant-skeleton-paragraph"
          >
            <li />
            <li />
            <li />
            <li
              style="width: 61%;"
            />
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`System/LdapConfig should get ldap setting from origin 2`] = `
<div>
  <div
    class="ant-card ant-card-bordered"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          system.title.ldap
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <section>
        <div
          class="ant-descriptions"
        >
          <div
            class="ant-descriptions-view"
          >
            <table>
              <tbody>
                <tr
                  class="ant-descriptions-row"
                >
                  <td
                    class="ant-descriptions-item"
                    colspan="3"
                  >
                    <div
                      class="ant-descriptions-item-container"
                    >
                      <span
                        class="ant-descriptions-item-label"
                      >
                        system.ldap.enableLdap
                      </span>
                      <span
                        class="ant-descriptions-item-content"
                      >
                        common.close
                      </span>
                    </div>
                  </td>
                </tr>
                <tr
                  class="ant-descriptions-row"
                >
                  <td
                    class="ant-descriptions-item"
                    colspan="3"
                  >
                    <div
                      class="ant-descriptions-item-container"
                    >
                      <span
                        class="ant-descriptions-item-content"
                      >
                        <button
                          class="ant-btn ant-btn-primary"
                          type="button"
                        >
                          <span>
                            common.modify
                          </span>
                        </button>
                      </span>
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </section>
      <form
        class="ant-form ant-form-horizontal"
        hidden=""
      >
        <div
          class="ant-form-item"
        >
          <div
            class="ant-row ant-form-item-row"
          >
            <div
              class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
            >
              <label
                class=""
                for="enable_ldap"
                title="system.ldap.enableLdap"
              >
                system.ldap.enableLdap
              </label>
            </div>
            <div
              class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-12 ant-col-md-10"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <button
                    aria-checked="false"
                    class="ant-switch"
                    id="enable_ldap"
                    role="switch"
                    type="button"
                  >
                    <div
                      class="ant-switch-handle"
                    />
                    <span
                      class="ant-switch-inner"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-form-item"
        >
          <div
            class="ant-row ant-form-item-row"
          >
            <div
              class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
            >
              <label
                class="ant-form-item-no-colon"
                title=" "
              >
                 
              </label>
            </div>
            <div
              class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-12 ant-col-md-10"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <div
                    class="ant-space ant-space-horizontal ant-space-align-center"
                  >
                    <div
                      class="ant-space-item"
                      style="margin-right: 8px;"
                    >
                      <button
                        class="ant-btn ant-btn-primary"
                        type="submit"
                      >
                        <span>
                          common.submit
                        </span>
                      </button>
                    </div>
                    <div
                      class="ant-space-item"
                    >
                      <button
                        class="ant-btn ant-btn-default"
                        type="button"
                      >
                        <span>
                          common.cancel
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`System/LdapConfig should update ldap setting after user update from 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          system.title.ldap
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <section
        hidden=""
      >
        <div
          class="ant-descriptions"
        >
          <div
            class="ant-descriptions-view"
          >
            <table>
              <tbody>
                <tr
                  class="ant-descriptions-row"
                >
                  <td
                    class="ant-descriptions-item"
                    colspan="3"
                  >
                    <div
                      class="ant-descriptions-item-container"
                    >
                      <span
                        class="ant-descriptions-item-label"
                      >
                        system.ldap.enableLdap
                      </span>
                      <span
                        class="ant-descriptions-item-content"
                      >
                        common.close
                      </span>
                    </div>
                  </td>
                </tr>
                <tr
                  class="ant-descriptions-row"
                >
                  <td
                    class="ant-descriptions-item"
                    colspan="3"
                  >
                    <div
                      class="ant-descriptions-item-container"
                    >
                      <span
                        class="ant-descriptions-item-content"
                      >
                        <button
                          ant-click-animating-without-extra-node="false"
                          class="ant-btn ant-btn-primary"
                          type="button"
                        >
                          <span>
                            common.modify
                          </span>
                        </button>
                      </span>
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </section>
      <form
        class="ant-form ant-form-horizontal"
      >
        <div
          class="ant-form-item ant-form-item-has-success"
        >
          <div
            class="ant-row ant-form-item-row"
          >
            <div
              class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
            >
              <label
                class=""
                for="enable_ldap"
                title="system.ldap.enableLdap"
              >
                system.ldap.enableLdap
              </label>
            </div>
            <div
              class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-12 ant-col-md-10"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <button
                    aria-checked="false"
                    class="ant-switch"
                    id="enable_ldap"
                    role="switch"
                    type="button"
                  >
                    <div
                      class="ant-switch-handle"
                    />
                    <span
                      class="ant-switch-inner"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-form-item"
        >
          <div
            class="ant-row ant-form-item-row"
          >
            <div
              class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
            >
              <label
                class="ant-form-item-no-colon"
                title=" "
              >
                 
              </label>
            </div>
            <div
              class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-12 ant-col-md-10"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <div
                    class="ant-space ant-space-horizontal ant-space-align-center"
                  >
                    <div
                      class="ant-space-item"
                      style="margin-right: 8px;"
                    >
                      <button
                        class="ant-btn ant-btn-primary"
                        type="submit"
                      >
                        <span>
                          common.submit
                        </span>
                      </button>
                    </div>
                    <div
                      class="ant-space-item"
                    >
                      <button
                        class="ant-btn ant-btn-default"
                        type="button"
                      >
                        <span>
                          common.cancel
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`System/LdapConfig should update ldap setting after user update from 2`] = `
<div>
  <div
    class="ant-card ant-card-loading ant-card-bordered"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          system.title.ldap
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <div
        class="ant-skeleton ant-skeleton-active"
      >
        <div
          class="ant-skeleton-content"
        >
          <ul
            class="ant-skeleton-paragraph"
          >
            <li />
            <li />
            <li />
            <li
              style="width: 61%;"
            />
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
